### PR TITLE
Fixed Issue #18581: Blocking users after X failed attempts counts incorrectly (off by 1)

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Review by Shnoulle — APPROVED** [review]

**Comment by olleharstedt** [comment]
Failing CI?

**Comment by gabrieljenik** [comment]
It is failing on a test which is not even impacted here.
Maybe re merging master?

BTW, what is the difference in between the push test and the pull_request test ?

File: tests/functional/acceptance/15532-em-warnings/ExpressionWarningsOnLogicTest.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public function testCheckInvalid()
         );
         try {
             self::$webDriver->get($url);
-            sleep(1);
+            sleep(5);
```
Comment (olleharstedt) [inline tests/functional/acceptance/15532-em-warnings/ExpressionWarningsOnLogicTest.php:56]: ???

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: tests/functional/acceptance/15532-em-warnings/ExpressionWarningsOnLogicTest.php (hunk 1 above)
Comment (gabrieljenik) [reply tests/functional/acceptance/15532-em-warnings/ExpressionWarningsOnLogicTest.php:56]: The test is failing. Making sure it wasn't a speed issue.

**Comment by Shnoulle** [comment]
> Updated the FailedLoginAttempt test. This still fails at ls\tests\ExpressionWarningsOnLogic --> Check invalid Will be back

Sometimes fail are really strange …

**Comment by gabrieljenik** [comment]
> > Updated the FailedLoginAttempt test. This still fails at ls\tests\ExpressionWarningsOnLogic --> Check invalid Will be back
> 
> Sometimes fail are really strange …

Will push a solution about it.

**Comment by olleharstedt** [comment]
First step should probably be to try to run the test steps manually, in the browser.

**Comment by gabrieljenik** [comment]
The issue is about some modal boxes that appear. We are working on it.
image
image

Maybe best is to set some test mode where we just don't show some stuff which can mess tests?
(modals, alerts)

**Comment by Shnoulle** [comment]
> Maybe best is to set some test mode where we just don't show some stuff which can mess tests?

Why this appear only now ? This 2 alert shown before ?
SSL alert can be disable by config

**Comment by Shnoulle** [comment]
😩

**Pull request closed without merging**